x-pack/plugins/apm/server/routes/storage_explorer/get_service_statistics.ts

```diff
@@ -0,0 +1,240 @@
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import { ApmPluginRequestHandlerContext } from '../typings';
+import { Setup } from '../../lib/helpers/setup_request';
+import {
+  IndexLifecyclePhaseSelectOption,
+  indexLifeCyclePhaseToDataTier,
+} from '../../../common/storage_explorer_types';
+import { getTotalTransactionsPerService } from './get_total_transactions_per_service';
+import {
+  PROCESSOR_EVENT,
+  SERVICE_NAME,
+  SERVICE_ENVIRONMENT,
+  TIER,
+  TRANSACTION_SAMPLED,
+  AGENT_NAME,
+  INDEX,
+} from '../../../common/elasticsearch_fieldnames';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { AgentName } from '../../../typings/es_schemas/ui/fields/agent';
+import {
+  getTotalIndicesStats,
+  getEstimatedSizeForDocumentsInIndex,
+} from './indices_stats_helpers';
+import { RandomSampler } from '../../lib/helpers/get_random_sampler';
+
+async function getMainServiceStatistics({
+  setup,
+  context,
+  indexLifecyclePhase,
+  randomSampler,
+  start,
+  end,
+  environment,
+  kuery,
+}: {
+  setup: Setup;
+  context: ApmPluginRequestHandlerContext;
+  indexLifecyclePhase: IndexLifecyclePhaseSelectOption;
+  randomSampler: RandomSampler;
+  start: number;
+  end: number;
+  environment: string;
+  kuery: string;
+}) {
+  const { apmEventClient } = setup;
+
+  const allIndicesStats = await getTotalIndicesStats({ context, setup });
+
+  const response = await apmEventClient.search('get_main_service_statistics', {
+    apm: {
+      events: [
+        ProcessorEvent.span,
+        ProcessorEvent.transaction,
+        ProcessorEvent.error,
+        ProcessorEvent.metric,
+      ],
+    },
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          filter: [
+            ...environmentQuery(environment),
+            ...kqlQuery(kuery),
+            ...rangeQuery(start, end),
+            ...(indexLifecyclePhase !== IndexLifecyclePhaseSelectOption.All
+              ? termQuery(
+                  TIER,
+                  indexLifeCyclePhaseToDataTier[indexLifecyclePhase]
+                )
+              : []),
+          ] as QueryDslQueryContainer[],
+        },
+      },
+      aggs: {
+        sample: {
+          random_sampler: randomSampler,
+          aggs: {
+            services: {
+              terms: {
+                field: SERVICE_NAME,
+                size: 500,
+              },
+              aggs: {
+                sample: {
+                  top_hits: {
+                    size: 1,
+                    _source: [AGENT_NAME],
+                    sort: {
+                      '@timestamp': 'desc',
+                    },
+                  },
+                },
+                indices: {
+                  terms: {
+                    field: INDEX,
+                    size: 500,
+                  },
+                  aggs: {
+                    number_of_metric_docs: {
+                      value_count: {
```

Comment:
why do you need a value_count agg? because it will use _doc_count? do you get a warning when running a terms agg on _index?

Reply:
On a transaction metric document, for example, `_doc_count` reflects the number of transactions that were aggregated